llama-index-integrations/retrievers/llama-index-retrievers-videodb/llama_index/retrievers/videodb/base.py

```diff
@@ -0,0 +1,88 @@
+from llama_index.core.base.base_retriever import BaseRetriever
+from llama_index.core.callbacks.base import CallbackManager
+from llama_index.core.schema import NodeWithScore, QueryBundle, TextNode
+
+import logging
+import os
+
+from typing import List, Optional
+from videodb import connect
+
+
+logger = logging.getLogger(__name__)
+
+
+class SearchType:
+    keyword = "keyword"
+    semantic = "semantic"
+
+
+class VideoDBRetriever(BaseRetriever):
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        collection: Optional[str] = "default",
+        video: Optional[str] = None,
+        score_threshold: Optional[float] = 0.2,
+        result_threshold: Optional[int] = 5,
+        search_type: Optional[str] = SearchType.semantic,
+        base_url: Optional[str] = None,
+        callback_manager: Optional[CallbackManager] = None,
+    ) -> None:
+        """Creates a new VideoDB Retriever."""
+        if api_key is None:
+            api_key = os.environ.get("VIDEO_DB_API_KEY")
+        if api_key is None:
+            raise Exception(
+                "No API key provided. Set an API key either as an environment variable (VIDEO_DB_API_KEY) or pass it as an argument."
+            )
+        self._api_key = api_key
+        self._base_url = base_url
+        self.video = video
+        self.collection = collection
+        self.score_threshold = score_threshold
+        self.result_threshold = result_threshold
+        self.search_type = search_type
+        super().__init__(callback_manager)
+
+    def _retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
```

Comment:
Do you have async client? Would be good to add `_aretrieve` if there is async client.

Reply:
@ravi03071991 We don't have async client right now.

Reply:
Okay. Cool. Thanks.